Message interviews:16 -  Read
From:    Mark Linton <linton@interviews.stanford.edu>
To:      <regler@control.lth.se>
Cc:      <interviews@interviews.stanford.edu>
Subject: Re:  GetCursor/SetCursor -- bug or feature

>I do not understand the asymmetric behaviour of GetCursor/SetCursor; I would
>prefer SetCursor(nil) to mean ``use parent's cursor again.''  Comments,
>please.  (I use InterViews 2.5 on a Sun-3, compiled with AT&T 1.2.1)

I agree, it's a bug.  The fix is to put the nil check in
Interactor::DoSetCursor in X11.c.  Below is the patch.

	Mark


--

Apply with ``cd iv; patch -p -N < diffs'' to patch files:

	src/InterViews/Text/terminal.h
	src/libtext/terminal.c

*** /tmp/,RCSt1a08357	Mon Jun 26 10:40:22 1989
--- X11.c	Mon Jun 26 10:09:49 1989
***************
*** 2336,2342 ****
  }

  void Interactor::DoSetCursor (Cursor* c) {
!     XDefineCursor(_world->display(), (Window)canvas->id, (XCursor)c->Id());
      Flush();
  }

--- 2336,2346 ----
  }

  void Interactor::DoSetCursor (Cursor* c) {
!     if (c == nil) {
! 	XUndefineCursor(_world->display(), (Window)canvas->id);
!     } else {
! 	XDefineCursor(_world->display(), (Window)canvas->id, (XCursor)c->Id());
!     }
      Flush();
  }

